base/server/healthcheck/pki/server/healthcheck/clones/plugin.py

```diff
@@ -0,0 +1,310 @@
+        self.clone_ocsps = []
+        self.master_tpss = []
+        self.clone_tpss = []
+        self.master_tkss = []
+        self.clone_tkss = []
+
+        self.security_domain = None
+        self.db_dir = None
+        self.subsystem_token = None
+        self.passwd = None
+
+        self.instance = PKIInstance(self.config.instance_name)
+
+    def contact_subsystem_using_pki(
+            self, subport, subhost, subsystemnick,
+            token_pwd, db_path, cmd, exts=None):
+        command = ["/usr/bin/pki",
+                   "-p", str(subport),
+                   "-h", subhost,
+                   "-n", subsystemnick,
+                   "-P", "https",
+                   "-d", db_path,
+                   "-c", token_pwd,
+                   cmd]
+
+        if exts is not None:
+            command.extend(exts)
+
+        output = None
+        try:
+            output = subprocess.check_output(command, stderr=subprocess.STDOUT)
+        except subprocess.CalledProcessError as e:
+            output = e.output.decode('utf-8')
+            return output
+
+        output = output.decode('utf-8')
+
+        return output
+
+    def contact_subsystem_using_sslget(
+            self, port, host, subsystemnick,
+            token_pwd, db_path, params, url):
+
+        command = ["/usr/bin/sslget"]
+
+        if subsystemnick is not None:
+            command.extend(["-n", subsystemnick])
+
+        command.extend(["-p", token_pwd, "-d", db_path])
+
+        if params is not None:
+            command.extend(["-e", params])
+
+        command.extend([
+            "-r", url, host + ":" + port])
+
+        logger.info(' command : %s ', command)
+        output = None
+        try:
+            output = subprocess.check_output(command, stderr=subprocess.STDOUT)
+        except subprocess.CalledProcessError as e:
+            output = e.output.decode('utf-8')
+            return output
+
+        output = output.decode('utf-8')
+
+        return output
+
+    def get_security_domain_data(self, host, port):
+        domain_data = None
+
+        try:
+            connection = PKIConnection(protocol='http',
+                                       hostname=host,
+                                       port=port,
+                                       verify=False)
+
+            securityDomainClient = SecurityDomainClient(connection)
+            domain_data = securityDomainClient.get_domain_info()
+
+        except BaseException as e:
+            logger.error("Internal server error %s", e)
+            return domain_data
+
+        systems = domain_data.subsystems
+        for s in systems.values():
+            for h in s.hosts.values():
+                if s.id == 'CA':
+                    if h.Clone == 'TRUE':
+                        self.clone_cas.append(h)
+                    else:
+                        self.master_cas.append(h)
+                    continue
+
+                if s.id == 'KRA':
```

Comment:
Should we use `elif`, in all conditions, to be little efficient?

Reply:
We can do that...

Reply:
Fixed.